packages/entities/entities-plugins/src/components/free-form/Datakit/DatakitForm.vue

```diff
@@ -148,11 +167,74 @@ function handleFlowConfigChange(newConfig: DatakitConfig, newUIData?: DatakitUID
 
 // Code editor
 
+function stringifyPath(path: Array<string | number>): string {
+  return path
+    .map((k) =>
+      typeof k === 'number'
+        ? `[${k}]`
+        : path.length && typeof k === 'string' && k.includes('.')
+          ? `["${k}"]`
+          : `.${k}`,
+    )
+    .join('')
+    .replace(/^\./, '')
+}
+
+function getSchemaErrorMessage(error: ZodError): string {
+  return error.issues
+    .map(
+      (issue) =>
+        `${stringifyPath(issue.path as Array<string | number>)} - ${
+          issue.message
+        }`,
+    )
+    .join('; ')
+}
+
+function handleCodeChange(newConfig: unknown) {
+  handleConfigChange(newConfig)
+
+  const { success, error } = DatakitConfigSchema.safeParse(newConfig)
+
+  const { success: compatSuccess } = DatakitConfigCompatSchema.safeParse(newConfig)
+  flowAvailable.value = compatSuccess
```

Comment:
q: Would it be possible for us to assume that a config passes the compat schema check if it passes the strict one?

Reply:
I think so 👍